templates/dj_dashboard/reservation/show.html.twig

```diff
@@ -40,19 +40,28 @@
 					</p>
 					<p class="mt-0">
 						&nbsp
-						{% for musicalStyle in reservation.musicalStyles %}
-							{{ musicalStyle.name }}
-							{% if not loop.last %},
-							{% endif %}
-						{% endfor %}
+						{% if reservation.musicalStyles is not empty %}
+							{% for musicalStyle in reservation.musicalStyles %}
+								{{ musicalStyle.name }}
+								{% if not loop.last %},
```

Comment:
regarde le join en Twig  https://twig.symfony.com/doc/2.x/filters/join.html
ça devrait t'éviter le bidouillage avec le not loop. last ;-)